src/controllers/project.controller.ts

```diff
@@ -118,14 +119,36 @@ export class ProjectController {
 		};
 
 		this.view.update({ type: MessageType.setSessionsSection, payload: sessionsViewObject });
-
 		if (this.selectedDeploymentId) {
 			await this.selectDeployment(this.selectedDeploymentId);
 		}
+
+		const lastDeployment = deployments ? deployments[deployments?.length - 1] : null;
```

Comment:
TICKET:

1. extract to separate function for re-use if needed, reduce function size, and separate the context
2. load only on user action

Reply:
https://linear.app/autokitteh/issue/UI-192/vscode-loadanddisplaydeployments-func-refactor
https://linear.app/autokitteh/issue/UI-193/vscode-singleshot-load-buildinfo-on-user-action